jujugui/static/gui/src/app/components/canvas-info/canvas-info.js

```diff
@@ -0,0 +1,210 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+/**
+  Displays information for usage and quickstart.
+  Shown when a charm or bundle is added to the canvas, and on deploy.
+*/
+class CanvasInfo extends React.Component {
+  constructor(props) {
+    super(props);
+    this.templateTags = {
+      details_link: {
+        onClick: this._handleViewDetails.bind(this),
+        content: '<span role="button" ' +
+          'class="link" ' +
+          'data-templatetag="details_link">View details</span>'
+      },
+      requires_cli_link: {
+        content: '<a ' +
+          'href="https://jujucharms.com/docs/stable/reference-install" ' +
+          'target="_blank">Juju CLI client</a>'
+      }
+    };
+    this.state= {
+      content: null,
+      displayName: null,
+      metadata: {}
+    };
+  }
+
+  componentDidMount() {
+    const entity = this.props.makeEntityModel(this.props.entity).toEntity();
+
+    this.setState({
+      displayName: entity.displayName
+    });
+
+    const files = this.props.entity.files;
+    if (files && files.indexOf('usage.md') !== -1) {
+      this.props.getFile(
+        this.props.entity.id,
+        'usage.md',
+        this._getUsageCallback.bind(this)
+      );
+    }
+  }
+
+  /**
+    Callback when usage.md is retrieved.
+    Even if there is an error we still want to display the basic box.
+
+    @param {Object} error Error from the API.
+    @param {String} usageContents The contents of the file.
+  */
+  _getUsageCallback(error, usageContents) {
+    if (error) {
+      console.error(error);
+    }
+    // Rather then parsing the JSON to read the error we check if the returned
+    // body starts with a '{'. If it does, it's not a markdown file so ignore.
+    if (usageContents && usageContents.substring(0, 1) !== '{') {
+      this.setState({
+        metadata: this.parseMarkdownMetadata(usageContents)
+      });
+      const markdown = this.props.marked(
```

Comment:
so marked won't fail with markdown that has a frontmatter block? If this is malformed does marked return an error? We should have some kind of error handling as pretty much anything can be in these markdown files.

A test for bad markdown would be nice as well.